### ISSUE TITLE
Add some explanation to example

### ISSUE THREAD

**Issue by homocomputeris** [JuliaDiffEq/DiffEqDevDocs.jl#13]
To help newcomers, it would be nice to have an explained and commented example of a new algorithm like the existing [Self-contained example](https://github.com/JuliaDiffEq/DiffEqDevDocs.jl/blob/master/docs/src/contributing/adding_algorithms.md#self-contained-example) which has some design points that are worth explaining:

- [ ] Why is alg_order defined outside its type?
- [ ] What exactly are caches supposed to hold? The example's field names suggest it's the Butcher tableau. (Not important, but why are they "caches" then?)
- [ ] What are `T1` and `T2` functions in the cache constructor? What operation do they perform on a method's coefficients?
- [ ] What does `alg_cache` do? What do its arguments mean? Why should it be called by hand?
- [ ] What should an `integrator` contain: what is `kshortsize`? what is the relation between `isfsal` and `fsalfirst`/`fsallast`?
- [ ] What entities should `initialize!`? Is it supposed to "pre-fill" all the known data like IVs/BCs, or something else?

**Comment by ChrisRackauckas** [comment]
Yeah, this is a good list of things to document. It should be pretty self-explanatory so if someone wants to take a stab at it I'd be willing to review it.

**Comment by homocomputeris** [comment]
@ChrisRackauckas then could you ask other contributors if they'd like to help with this or invite them here?
It seems that this repo is less popular than the main ones.

**Comment by ChrisRackauckas** [comment]
Yeah, maybe @kanav99 can help out. Or it can be a first task for one of the next GSoCs.

**Comment by kanav99** [comment]
I would love to see some great documentation that explains the whole control flow in OrdinaryDiffEq. Best way should be to let a new student explore the repository and log down things as he/she may observe and finally make a PR. In any case, I can give this issue a stab myself too.

**Comment by kanav99** [comment]
We should also mention SimpleDiffEq.jl somewhere too, that's where I understood the complete flow from.

**Comment by ChrisRackauckas** [comment]
Yeah, next GSoC would be a good time. I usually leave around little things like this as teaching exercises.
